aidial_adapter_openai/utils/tokens.py

```diff
@@ -82,10 +82,11 @@ def discard_messages(
         len(kept_messages) == system_messages_count
         and system_messages_count != n
     ):
-        raise HTTPException(
+        raise DialException(
             message=f"The token size of system messages and the last user message ({prompt_tokens}) exceeds prompt token limit ({max_prompt_tokens})",
             status_code=400,
             type="invalid_request_error",
+            display_message="You exceeded amount of tokens for this chat. Please try create another one.",
```

Comment:
`The token size of system messages and the last user message ({prompt_tokens}) exceeds prompt token limit ({max_prompt_tokens}). Try reducing the length of the messages.`

Reply:
Same as above:

```python
    # Then non-system messages in the reverse order
    for idx, message in reversed(list(enumerate(messages))):
        if message["role"] != "system":
            prompt_tokens += tokenizer.calculate_tokens_per_message(message)

            if max_prompt_tokens < prompt_tokens:
                break

            kept_messages.add(idx)

    if (
        len(kept_messages) == system_messages_count
        and system_messages_count != n
    ):
        raise DialException(
            message=f"The token size of system messages and the last user message ({prompt_tokens}) exceeds prompt token limit ({max_prompt_tokens})",
            status_code=400,
            type="invalid_request_error",
            display_message="You exceeded amount of tokens for this chat. Please try create another one.",
        )
```

That check basically says, that none of the user messages was added, before we exceeded the limit.

I don't think reducing size of messages can help here at all.

Reply:
As discussed, a user has an ability to modify system message for the existing chat conversation in DIAL Chat.